6task6/src/index.js

```diff
@@ -0,0 +1,366 @@
+    }
+    return "incorrect input";
+}
+
+
+function validateAd(item) {
+    if (!item) {
+        return false;
+    }
+    if (item.id === "" || typeof item.id !== "string")
+        return false;
+
+    if (item.description === "" || typeof item.description !== "string" || item.description.length > 200)
+        return false;
+
+    if (item.link === "" || typeof item.link !== "string")
+        return false;
+
+    if (item.discount === "" || typeof item.discount !== "string")
+        return false;
+
+    if (item.vendor === "" || typeof item.vendor !== "string")
+        return false;
+
+    if (item.createdAt instanceof Date || item.createdAt !== "")
+        return true;
+
+    if (item.validUntil instanceof Date || item.validUntil !== "")
+        return true;
+
+    if (item.hashTags.length == 0) {
+        return true;
+    } else
+        for (let i = 0; i < item.hashTags.length; i++) {
+            if (!(item.hashTags[i].substring(0, 1) === '#')) {
+                return false;
+            }
+        }
+    return true;
+}
+
+
+    function getAds(skip, top, filterConfig) {
+        let result;
+        if (!filterConfig) {
+            result = adList;
+        }
+
+        if (filterConfig) {
+
+            if (filterConfig.vendor) {
+                result = adList.filter(function (item) {
+                        return item.vendor.toLowerCase().indexOf(filterConfig.vendor.toLowerCase()) + 1;
+                    }
+                );
+            }
+
+            if (filterConfig.createdAt) {
+                result = adList.filter(function (item) {
+                    return item.createdAt >= filterConfig.createdAt;
+                });
+            }
+            if (filterConfig.validUntil) {
+                result = adList.filter(function (item) {
+                    return item.validUntil.getTime() <= filterConfig.validUntil.getTime();
+                });
+            }
+
+        }
+        return result.sort((function (a, b) {
+                return a.createdAt - b.createdAt
+            }
+        )).slice(skip, skip + top);
+    }
+
+    function addAd(item) {
+        const date = new Date();
+        item.id = +date;
+        item.createdAt = new Date();
+        if (validateAd(item)) {
+            adList.push(item);
+            return true;
+        }
+        return false;
+    }
+
+
+
+    function editAd(id, adItem) {
+
+    var num = adList.findIndex(item => item.id === id);
+
+    if (adItem.description) {
+        adList[num].description = adItem.description;
+    }
+    if (adItem.link) {
+        adList[num].link = adItem.link;
+    }
+    if (adItem.discount) {
+        adList[num].discount = adItem.discount;
+    }
+    if (adItem.validUntil) {
+        adList[num].validUntil = adItem.validUntil;
+    }
+    if (adItem.hashTags) {
+        adList[num].hashTags = adItem.hashTags;
+    }
+    if (validateAd(adList[num]) === false){
+        return false;
```

Comment:
I'd rather advice you to copy existing object, change it and add it to array only if resulting ad object is valid - that's safer.